typescript/create-onchain-agent/templates/next/app/api/agent/evm/privy/route.ts

```diff
@@ -105,25 +106,30 @@ async function getOrInitializeAgent(): Promise<ReturnType<typeof createReactAgen
     const walletProvider = await PrivyWalletProvider.configureWithWallet(config);
 
     // Initialize AgentKit: https://docs.cdp.coinbase.com/agentkit/docs/agent-actions
-    const agentkit = await AgentKit.from({
-      walletProvider,
-      actionProviders: [
-        wethActionProvider(),
-        pythActionProvider(),
-        walletActionProvider(),
-        erc20ActionProvider(),
-        // The CDP API Action Provider provides faucet functionality on base-sepolia. Can be removed if you do not need this functionality.
+    const actionProviders: ActionProvider[] = [
+      wethActionProvider(),
+      pythActionProvider(),
+      walletActionProvider(),
+      erc20ActionProvider(),
+    ];
+    const canUseCdpApi = process.env.CDP_API_KEY_NAME && process.env.CDP_API_KEY_PRIVATE_KEY;
+    if (canUseCdpApi) {
```

Comment:
not blocking, but this feels like a smell to me. It seems like logic that should live in the CLI if possible, and not in the produced code. WDYT?

Reply:
Limitation of the `cli` at this moment. I have a backlog ticket to refactor it to use a proper templating engine, which would give me the flexibility to handle this appropriately.

I'm hoping to prioritize that refactor early next week so I can clean up any branching logic in the routes